src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,170 @@
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> removedNodeByValue = findNodeByValue(object);
+        if (removedNodeByValue == null) {
+            return false;
+        } else {
+            unlink(removedNodeByValue);
+            size--;
+            return true;
+        }
+    }
+
+    private Node<T> findNodeByValue(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(currentNode.element, object)) {
+                return currentNode;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndexRange(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index: "
+                                                        + index
+                                                        + " out of bounds "
+                                                        + "for size "
+                                                        + size);
+        }
+    }
+
+    private void checkIndexRangeInAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: "
+                                                        + index
+                                                        + " out of bounds "
+                                                        + "for size "
+                                                        + size);
+        }
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        if (index < size / 2) {
+            Node<T> foundNodeByIndexFromHead = head;
+            for (int i = 0; i < index; i++) {
+                foundNodeByIndexFromHead = foundNodeByIndexFromHead.next;
+            }
+            return foundNodeByIndexFromHead;
+        } else {
+            Node<T> foundNodeByIndexFromTail = tail;
+            for (int i = size - 1; i > index; i--) {
+                foundNodeByIndexFromTail = foundNodeByIndexFromTail.prev;
+            }
+            return foundNodeByIndexFromTail;
+        }
+    }
+
+    private void unlink(Node<T> node) {
+        if (node.prev == null) {
+            head = head.next;
+        } else if (node.next == null) {
+            tail = tail.prev;
+        } else {
+            node.prev.next = node.next;
+            node.next.prev = node.prev;
+        }
+    }
+
+    private class Node<E> {
+        private E element;
+        private Node<E> prev;
+        private Node<E> next;
+
+        private Node(Node<E> prev, E element, Node<E> next) {
+            this.element = element;
+            this.next = next;
+            this.prev = prev;
+        }
     }
```

Comment:
Getters and setters are not necessary for the `Node` class since it's an inner class and we have access to private fields of the inner class. Also, ensure that private methods and classes are placed after public ones.